10_rest/willshen8/cmd/puppy-server/main_test.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/anz-bank/go-course/10_rest/willshen8/pkg/puppy"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Unit test for parseCmdArgs() with long flags
+func TestParseCmdArgs(t *testing.T) {
+	args = []string{"--data", "./../../data/puppy.json", "--port", "1234", "--store", "map"}
+	returnedArgs := parseCmdArgs(args)
+	assert.Equal(t, "./../../data/puppy.json", returnedArgs.file)
+	assert.Equal(t, 1234, returnedArgs.port)
+	assert.Equal(t, "map", returnedArgs.storer)
+}
+
+// Unit test for parseCmdArgs() with short flags
+func TestShortFlag(t *testing.T) {
+	args = []string{"-d", "./../../data/puppy.json", "-p", "1234", "-s", "map"}
+	returnedArgs := parseCmdArgs(args)
+	assert.Equal(t, "./../../data/puppy.json", returnedArgs.file)
+	assert.Equal(t, 1234, returnedArgs.port)
+	assert.Equal(t, "map", returnedArgs.storer)
+}
+
+// Unit test (positive test) for parsePortFlag()
+func TestParseCorrectPort(t *testing.T) {
+	// setup the test server with arguments
+	args = []string{"--data", "./../../data/puppy.json", "--port", "1234"}
+	parsedArgs := parseCmdArgs(args)
+	flagReturned, _ := parsePortFlag(parsedArgs)
+	var expectedFlag = "1234"
+	assert.Equal(t, expectedFlag, flagReturned)
+}
+
+// Unit test (negative test) for parsePortFlag()
+func TestParseWrongPort(t *testing.T) {
+	// setup the test server with arguments
+	args = []string{"--data", "./../../data/puppy.json", "--port", "123456789"}
+	parsedArgs := parseCmdArgs(args)
+	flagReturned, flagError := parsePortFlag(parsedArgs)
+	var expectedFlag = "0"
+	var expectedError = "Invalid port number entered."
+	assert.Equal(t, expectedFlag, flagReturned)
+	assert.Equal(t, expectedError, flagError.Error())
+
```

Comment:
Unnecessary blank line.
No need for blank lines at beginning or end of block.
Use blank lines between type, method and function definitions and within a block for logical separation.

Reply:
fixed.